app/app.h

```diff
@@ -1,8 +1,49 @@
 #pragma once
 #include <string>
 #include <memory>
+#include <iostream>
 
 namespace vsite::oop::v4 
 {
+	class animal {
+	public:
+		virtual ~animal()=default;
+		virtual std::string species() const=0;
+		virtual unsigned legs() const=0;
+	};
+
+	class insect : public animal {};
```

Comment:
Ideja ove tri klase je da implementiraju ono što je zajedničko svim pticama/insektima/paucima - broj nogu (da se ne mora u klasi za svaku moguću pticu ponovo pisati funkcija `legs`)